pkg/reconciler/v1alpha1/taskrun/taskrun_test.go

```diff
@@ -945,3 +972,207 @@ func TestCreateRedirectedBuild(t *testing.T) {
 		t.Errorf("services accounts do not match: %s should be %s", b.Spec.ServiceAccountName, tr.Spec.ServiceAccount)
 	}
 }
+
+func TestTaskRunWithTaskSpec(t *testing.T) {
```

Comment:
probably dont need the extra case in test test above if we have this (or vice versa)

Reply:
maybe prefer adding cases to the existing test so we dont need to duplicate the assertions?